autogen.sh

```diff
@@ -22,10 +44,14 @@ echo "=== done === "
 echo
 
 
+########################################################################
+## Generating required files
+########################################################################
+
 # backend pup functions
 for x in seq cuda ; do
     echo_n "generating backend pup functions for ${x}... "
-    ./src/backend/${x}/genpup.py
+    ./src/backend/${x}/genpup.py --pup-max-nesting=${pup_max_nesting}
```

Comment:
Would it be simpler if you use the same option syntax for `autogen.sh` and `genpup.py`? You can use a case switch and collect the arguments as is and pass them to `genpup.py`.

Reply:
For now, the arguments are only for `genpup.py`, but `autogen.sh` runs more than that one script (for example, `gentests.py`).  So we can't simply pass all arguments to `genpup.py`.

Reply:
I understand. I mean something like this:
```
genpup_args=
for a in $@ ; do
    case a in 
        --pup-*) genpup_args="$genpup_args $a" ;;
    esac
done
```
then
```
    eval ./src/.../genpup.py $genpup_args
```

Reply:
Yeah, that's a good idea.  It'll also naturally separate the namespace for the arguments for different scripts.  I'll fix it.